pkg/rules/rules.go

```diff
@@ -21,14 +21,23 @@ import (
 	"github.com/timescale/promscale/pkg/rules/adapters"
 )
 
-type manager struct {
+type Manager interface {
```

Comment:
Having something called Manager and it having this large interface is a smell. 

Not to hold up progress too much, I would at least have a TODO here to refactor this part of the code.